dbms/src/Flash/Coprocessor/DAGQueryBlockInterpreter.cpp

```diff
@@ -185,10 +187,19 @@ void DAGQueryBlockInterpreter::handleTableScan(const TiDBTableScan & table_scan,
 {
     const auto push_down_filter = PushDownFilter::pushDownFilterFrom(query_block.selection_name, query_block.selection);
 
-    DAGStorageInterpreter storage_interpreter(context, table_scan, push_down_filter, max_streams);
-    storage_interpreter.execute(pipeline);
+    if (context.isDisaggregatedComputeMode())
+    {
+        StorageDisaggregatedInterpreter disaggregated_tiflash_interpreter(context, table_scan, push_down_filter, max_streams);
```

Comment:
Can we have some abstraction between `StorageDisaggregatedInterpreter` and `DAGStorageInterpreter`, e.g.: Either having them inherit a same parent class, or combining them into one class and only differentiate the two modes using a flag like `is_disaggregated`.

That way, `interpreter.execute()` and `analyzer = ...` calls don't have to duplicate for each case.

Reply:
1. Parent class: Its only member is `analyzer`, ant its only member function is like `getAnalyzer()`. It's a little strange.
2. Add flag: It means we still need to construct all memebers of `DAGStraogeInterpreter`, but most of them is useless for `StorageDisaggregatedInterpreter `